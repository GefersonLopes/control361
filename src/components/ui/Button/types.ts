import type { ButtonHTMLAttributes, ReactNode } from "react";

export type ButtonProps = {
  variant?: "primary" | "secondary" | "tertiary";
  size?: "sm" | "md" | "lg";
  children: ReactNode;
} & ButtonHTMLAttributes<HTMLButtonElement>;
