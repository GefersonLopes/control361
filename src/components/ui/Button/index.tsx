import clsx from "clsx";

import type { ButtonProps } from "./types";

export default function Button({
  variant = "primary",
  size = "md",
  className,
  children,
  ...props
}: ButtonProps) {
  const base =
    "rounded-lg font-medium focus:outline-none focus-visible:ring text-poppins";
  const variants: Record<string, string> = {
    primary:
      "bg-primary text-light hover:bg-primary/80 focus-visible:ring-primary/60",
    secondary:
      "bg-secondary text-light hover:bg-secondary/80 focus-visible:ring-secondary/60",
    tertiary:
      "bg-tertiary text-light hover:bg-tertiary/80 focus-visible:ring-tertiary/60",
  };
  const sizes: Record<string, string> = {
    sm: "px-3 py-1 text-sm",
    md: "px-5 py-1.5 text-sm",
    lg: "px-6 py-2 text-base",
  };

  return (
    <button
      className={clsx(base, variants[variant], sizes[size], className)}
      {...props}
    >
      {children}
    </button>
  );
}
