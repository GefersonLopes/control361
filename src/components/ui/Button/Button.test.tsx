import "@testing-library/jest-dom";

import { fireEvent, render, screen } from "@testing-library/react";

import Button from "../Button";

describe("Button component – testes adicionais", () => {
  it("fica desabilitado quando `disabled` é true e não dispara onClick", () => {
    const handleClick = jest.fn();
    render(
      <Button disabled onClick={handleClick}>
        Não posso
      </Button>,
    );
    const btn = screen.getByRole("button", { name: /não posso/i });
    expect(btn).toBeDisabled();
    fireEvent.click(btn);
    expect(handleClick).not.toHaveBeenCalled();
  });

  it("usa `type='button'` por padrão e permite sobrescrever para `submit`", () => {
    render(<Button>Default</Button>);
    expect(screen.getByRole("button")).toHaveAttribute("type", "button");

    render(<Button type="submit">Enviar</Button>);
    expect(screen.getByRole("button", { name: /enviar/i })).toHaveAttribute(
      "type",
      "submit",
    );
  });

  it("propaga atributos HTML arbitrários (`aria-label`, `data-*`, etc.)", () => {
    render(
      <Button aria-label="Fechar diálogo" data-testid="btn-123">
        X
      </Button>,
    );
    const btn = screen.getByTestId("btn-123");
    expect(btn).toHaveAttribute("aria-label", "Fechar diálogo");
    expect(btn).toHaveTextContent("X");
  });

  it("combina com o snapshot", () => {
    const { asFragment } = render(
      <Button variant="tertiary" size="sm">
        Snap
      </Button>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
