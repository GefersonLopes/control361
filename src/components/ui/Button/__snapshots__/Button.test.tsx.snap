// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component â€“ testes adicionais combina com o snapshot 1`] = `
<DocumentFragment>
  <button
    class="rounded-lg font-medium focus:outline-none focus-visible:ring text-poppins bg-tertiary text-light hover:bg-tertiary/80 focus-visible:ring-tertiary/60 px-3 py-1 text-sm"
    type="button"
  >
    Snap
  </button>
</DocumentFragment>
`;
